primitives/src/locktime/relative.rs

```diff
@@ -251,60 +241,114 @@ impl LockTime {
+    /// > timespan starts from the median-time-past of the output’s previous block, and ends at the
+    /// > MTP of the previous block.
+    /// >
+    /// > ... a relative time-based lock-time n can be included into any block produced 512 * n
+    /// > seconds after the mining date of the output it is spending, or any block thereafter. The
+    /// > mining date of the output is equal to the median-time-past of the previous block which
+    /// > mined it.
+    ///
+    /// > The block produced time is equal to the median-time-past of its previous block.
+    ///
+    /// # Parameters
+    ///
+    /// * `prevout_mining_date`: The mining date of the output (defined in quote above).
+    /// * `chain_mtp`: The MTP of the previous block (defined in quote above).
+    ///
+    /// Both parameters are UNIX timestamps.
+    ///
     /// # Errors
     ///
     /// Returns an error if this lock is not lock-by-time.
     ///
     /// # Examples
     ///
     /// ```rust
-    /// # use bitcoin_primitives::Sequence;
-    /// # use bitcoin_primitives::locktime::relative::{LockTime, Height, Time};
+    /// use bitcoin_primitives::Sequence;
+    /// use units::locktime::absolute::Time;
     ///
-    /// let intervals: u16 = 70; // approx 10 hours;
-    /// let lock = Sequence::from_512_second_intervals(intervals).to_relative_lock_time().expect("valid time");
-    /// assert!(lock.is_satisfied_by_time(Time::from_512_second_intervals(intervals + 10)).expect("a time"));
+    /// // Units of 512 seconds that the output we want to spend has to wait.
+    /// let span: u16 = 100;
+    ///
+    /// // Arbitrary UNIX timestamp representing the MTP of the block that created the prevout we want to spend.
+    /// let prevout_mining_date = Time::from_consensus(1_000_000_000).expect("valid time");
+    ///
+    /// // A unix timestamp 512 seconds after the timespan since the MTP of the prevout block.
+    /// let mtp_of_tip_of_chain = Time::from_consensus(1_000_000_000 + 512 * span as u32).expect("valid time");
+    ///
+    /// let lock = Sequence::from_512_second_intervals(span).to_relative_lock_time().expect("valid interval");
+    /// assert!(lock.is_satisfied_by_time(prevout_mining_date, mtp_of_tip_of_chain).unwrap());
     /// ```
     #[inline]
     #[cfg_attr(all(test, mutate), mutate)]
-    pub fn is_satisfied_by_time(&self, time: Time) -> Result<bool, IncompatibleTimeError> {
+    pub fn is_satisfied_by_time(
+        &self,
+        prevout_mining_date: Time,
+        chain_mtp: Time,
+    ) -> Result<bool, IncompatibleTimeError> {
         use LockTime::*;
 
         match *self {
-            Time(ref t) => Ok(t.value() <= time.value()),
-            Blocks(height) => Err(IncompatibleTimeError { time, height }),
+            Time(ref t) => {
+                let required_time =
+                    prevout_mining_date.to_consensus_u32() + u32::from(t.value() * 512);
```

Comment:
Perhaps we should add a function to `primitives::locktime::relative::Time` to calculate this

```rust

    /// Returns the number of seconds this `Time` encodes i.e., the timespan.
    #[inline]
    pub fn seconds(self) -> u32 { self.0 * 512 }
```

(Note I used your new nomenclature in the docs :)

Reply:
GI